File: DataEncryptionKeyGenerator.js
const fs = require('fs');
const base64 = require('uuid-base64');
const mongodb = require('mongodb');
const { ClientEncryption } = require('mongodb-client-encryption')(mongodb);

const { MongoClient } = mongodb;

const connectionString = 'mongodb://localhost:27017';
const keyVaultNamespace = 'encryption.__keyVault';
const path = './master-key.txt';

let kmsProviders;

try {
  const data = fs.readFileSync(path);
  kmsProviders = {
    local: {
      key: data,
    },
  };
} catch(err) {
  console.error(err);
}

const client = new MongoClient(connectionString, {
  useNewUrlParser: true,
  useUnifiedTopology: true,
});

async function DataEncryptionKeyGenerator() {
  try {
    await client.connect();
    const encryption = new ClientEncryption(client, {
      keyVaultNamespace,
      kmsProviders,
    });
    const key = encryption.createDataKey('local');
    const base64DataKeyId = key.toString('base64');
    const uuidDataKeyId = base64.decode(base64DataKeyId);
    console.log('DataKeyId [UUID]: ', uuidDataKeyId);
    console.log('DataKeyId [base64]: ', base64DataKeyId);
  } finally {
    await client.close();
  }
}
DataEncryptionKeyGenerator();

File: data_encryption_key_generator.py
import base64

from pymongo import MongoClient
from pymongo.encryption_options import AutoEncryptionOpts
from pymongo.encryption import ClientEncryption
from bson.codec_options import CodecOptions
from bson.binary import Binary, STANDARD, UUID

connection_string = "mongodb://localhost:27017"
key_vault_namespace = "encryption.__keyVault"

path = "./master-key.txt"
local_master_key = Binary(open(path, "rb").read(96))

kms_providers = {
    "local": {
        "key": local_master_key,
    },
}

fle_opts = AutoEncryptionOpts(
    kms_providers,
    key_vault_namespace,
    mongocryptd_bypass_spawn=True
)

client = MongoClient(
    connection_string,
    auto_encryption_opts=fle_opts
)

with ClientEncryption(
    kms_providers,
    key_vault_namespace,
    client,
    CodecOptions(uuid_representation=STANDARD)
) as client_encryption:

    data_key = client_encryption.create_data_key("local")

uuid_data_key_id = UUID(bytes=data_key)
base_64_data_key_id = base64.b64encode(data_key).decode("utf-8")

print("DataKeyId [UUID]: ", uuid_data_key_id)
print("DataKeyId [base64]: ", base_64_data_key_id)
